src/librustc_mir/build/matches/test.rs

```diff
@@ -76,23 +63,33 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
                 } else {
                     BinOp::Eq
                 };
-                Test {
-                    span: match_pair.pattern.span,
-                    kind: TestKind::Len { len: len as u64, op: op },
-                }
+                Some(TestKind::Len { len: len as u64, op: op })
             }
 
+            PatternKind::Or { .. } |
             PatternKind::AscribeUserType { .. } |
             PatternKind::Array { .. } |
             PatternKind::Wild |
             PatternKind::Binding { .. } |
             PatternKind::Leaf { .. } |
             PatternKind::Deref { .. } => {
-                self.error_simplifyable(match_pair)
+                None
```

Comment:
My current plan is to lower each "subpattern" here and create the `TestKind::Or` variant that will contain the `TestKind`'s for the "subpatterns"